File: python/003.py
def isPalindrome(s):
    left, right = 0, len(s) - 1
    while left < right:
        if not s[left].isalnum():
            left+=1
            continue
        if not s[right].isalnum():
            right-=1
            continue
        if s[left].lower() != s[right].lower():
            return False
        left += 1
        right -= 1
    return True


def isAlphanumeric(char):
    return char >= 'a' and char <= 'z' or char >= 'A' and char <= 'Z' or char >= '0' and char <= '9' 

def isSame(char1, char2):
    return char1.lower() == char2.lower()

print(isPalindrome("0race###Car0....."))

File: golang/003.go
package main

import "fmt"

func isPalindrome(s string) bool {
	leftPointer := 0
	rightPointer := len(s)-1

	for leftPointer <= rightPointer{
		if !(s[leftPointer] >= 'a' && s[leftPointer] <= 'z' || s[leftPointer] >= 'A' && s[leftPointer] <= 'Z'){
			leftPointer++
			continue
		}
		if !(s[rightPointer] >= 'a' && s[rightPointer] <= 'z' || s[rightPointer] >= 'A' && s[rightPointer] <= 'Z'){
			rightPointer--
			continue
		}
		if !isSame(s[leftPointer], s[rightPointer]){
			return false
		}
		leftPointer++
		rightPointer--		
	}
	return true
}

func isSame(char1, char2 byte) bool {
	return char1 == char2
}

func main(){
	words := []string{"hello", "helo","raccar"}
	for _, word := range words{
		fmt.Println(isPalindrome(word))
	}
}